test/mainnet/ynbnbk.spec.sol

```diff
@@ -79,15 +74,62 @@ contract VaultMainnetYnBNBkTest is Test, AssertUtils, MainnetActors {
         uint256 assetBalance = IERC20(assetAddress).balanceOf(address(vault));
         assertEq(assetBalance, assets, "Vault should hold the deposited assets");
 
+        assertEq(vault.balanceOf(bob), depositedShares, "bob should have the correct amount of shares");
+
+        uint256 oldTotalAssets = vault.totalAssets();
+        assertApproxEqAbs(
+            oldTotalAssets,
+            previousTotalAssets + stakeManager.convertSnBnbToBnb(assets),
+            1000,
+            "New total assets should equal deposit amount plus original total assets"
+        );
+
+        assertEq(address(vault.provider()), MC.PROVIDER, "provider should match");
+
+        assertEq(
+            stakeManager.convertSnBnbToBnb(1e18),
+            IProvider(MC.PROVIDER).getRate(MC.SLISBNB),
+            "slisBNB rate should match"
+        );
+
+        uint256 beforeBalance = IERC20(assetAddress).balanceOf(address(MC.YNBNBk));
+
         vm.startPrank(PROCESSOR);
+
         processApproveAsset(assetAddress, assets, MC.YNBNBk);
         processDepositYnBNBk(assets);
 
-        uint256 ynBnbkBalance = IERC20(MC.YNBNBk).balanceOf(address(vault));
         vm.stopPrank();
 
         uint256 newTotalAssets = vault.totalAssets();
-        assertApproxEqAbs(newTotalAssets, previousTotalAssets + stakeManager.convertSnBnbToBnb(assets), 100, "New total assets should equal deposit amount plus original total assets");
+
+        assetBalance = IERC20(assetAddress).balanceOf(address(vault));
+        assertEq(assetBalance, 0, "Vault should hold the deposited assets");
+
+        assetBalance = IERC20(assetAddress).balanceOf(address(MC.YNBNBk));
+        assertEq(assetBalance, beforeBalance + assets, "Vault should hold the deposited assets");
+
+        uint256 previewShares = Vault(payable(MC.YNBNBk)).previewDeposit(assets);
+        uint256 actualShares = Vault(payable(MC.YNBNBk)).balanceOf(address(vault));
+
+        assertEqThreshold(previewShares, actualShares, 10, "previewShares should equal actualShares");
+
+        uint256 previewAssets = Vault(payable(MC.YNBNBk)).previewRedeem(actualShares);
+
+        assertApproxEqAbs(
+            newTotalAssets,
+            previousTotalAssets + previewAssets,
+            1000,
+            "New total assets should equal to previous total assets plus previewAssets"
+        );
+
+        // TODO: this should be true
```

Comment:
@danoctavian Could you take a closer look at this test? 

The provider returns different values for rates for `ynBNBk` and `slisBNB`, and so the totalAssets value changes upon deposit into ynBNBk from ynBNBx. 

We may need to introduce a fix if you think this is important.

Reply:
let's fix this, it's normaly ynBNBk is not 1:1 with slisBNB, it should use its own rate.

ynBNBk is yield token .

Reply:
this is pre-upgrade as discussed